Source/Core/DolphinQt2/MenuBar.cpp

```diff
@@ -271,3 +309,58 @@ void MenuBar::InstallWAD()
 
   result_dialog.exec();
 }
+
+void MenuBar::PerformOnlineUpdate(const std::string& region)
+{
+  const int confirm = QMessageBox::question(
+      nullptr, tr("Confirm"), tr("Connect to the Internet and perform an online system update?"));
+  if (confirm != QMessageBox::Yes)
+    return;
+
+  QProgressDialog dialog;
+  dialog.setCancelButton(nullptr);
+  dialog.setLabelText(tr("Preparing to update...\nThis can take a while. Do not close Dolphin."));
+  dialog.setWindowTitle(tr("Updating"));
+  dialog.setWindowModality(Qt::WindowModal);
+  dialog.setWindowFlags(dialog.windowFlags() & ~Qt::WindowCloseButtonHint);
+  dialog.setMinimumSize(360, 120);
+
+  std::future<WiiUtils::UpdateResult> result = std::async(
+      std::launch::async, &WiiUtils::DoOnlineUpdate,
+      [&dialog](size_t processed, size_t total, u64 title_id) {
+        Core::QueueHostJob(
+            [&dialog, processed, total, title_id]() {
+              dialog.setRange(0, static_cast<int>(total));
+              dialog.setValue(static_cast<int>(processed));
+              dialog.setLabelText(
+                  tr("Updating title %1...\nThis can take a while. Do not close Dolphin.")
+                      .arg(title_id, 16, 16, QLatin1Char('0')));
+            },
+            true);
+      },
+      region);
+
+  dialog.exec();
+  while (result.wait_for(std::chrono::milliseconds(50)) != std::future_status::ready)
+    qApp->processEvents(QEventLoop::ExcludeUserInputEvents);
+
+  switch (result.get())
+  {
+  case WiiUtils::UpdateResult::AlreadyUpToDate:
+    QMessageBox::information(nullptr, tr("Update completed"),
+                             tr("The emulated Wii console is already up-to-date."));
+    return;
+  case WiiUtils::UpdateResult::Failed:
+    QMessageBox::critical(nullptr, tr("Update failed"),
+                          tr("Could not perform an online update. Please check your Internet "
+                             "connection and try again."));
+    return;
+  case WiiUtils::UpdateResult::Succeeded:
+    QMessageBox::information(nullptr, tr("Update completed"),
+                             tr("The emulated Wii console has been updated."));
+    break;
+  }
+
+  UpdateToolsMenu(false);
```

Comment:
It's to refresh the enabled/disabled menu items, and hopefully soon enough, the label for the "Start Wii System Menu" action. (See [DolphinWX](https://github.com/dolphin-emu/dolphin/blob/b2af07a/Source/Core/DolphinWX/MainMenuBar.cpp#L580).)